serenity-model/src/main/java/net/thucydides/core/requirements/FileSystemRequirementsTagProvider.java

```diff
@@ -120,7 +120,7 @@ public FileSystemRequirementsTagProvider(String rootDirectory) {
         this(filePathFormOf(rootDirectory), Injectors.getInjector().getProvider(EnvironmentVariables.class).get());
     }
 
-    private final ReentrantLock requirementsLock = new ReentrantLock();
+    private final Object requirementsLock = new Object();
```

Comment:
The ReentrantLock class is intended to be a more flexible replacement of the synchronised keyword introduced in Java 5 - is there a reason you suspect this is not working as intended? I would prefer to keep the ReentrantLock approach unless there is a compelling reason that this isn't working.

Reply:
For one: the unlock is not a finally block, so won't be executed if an exception is thrown.
A synchronized block is much easier in this aspect - i.e. you just cannot use it in a way that would forget to release it again.

Reply:
Second: unless you need the flexibility of Java 5 locks, you could (and should IMO) just use synchronized as well, because they're simpler to use